text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,1251 @@
+  └  • Action: 'Add' (expected: 'Import')
+
+Error: Construct upgrade validation failed
+```
+
+**Example 2:** If the change set analysis detects the existing table does not have policy action `Retain` (by setting
+the `RemovalPolicy` on existing table construct), the CLI will show:
+
+```s
+Performing construct upgrade validation...
+
+⚠️  Note: This is an experimental feature
+
+Validations:
+✅ No CloudFormation drift detected
+❌ Change set analysis failed
+  │ Resource: MyTable794EDED1 (AWS::DynamoDB::Table)
+  └   • RemovalPolicy: 'Delete' (expected: 'Retain')
+
+Error: Construct upgrade validation failed
+```
+
+**Example 3:** If the change set analysis detects the Replica custom resource does not have policy action `Retain`
+(by setting the `ReplicaRemovalPolicy` on existing table construct), the CLI will show:
+
+```s
+Performing construct upgrade validation...
+
+⚠️  Note: This is an experimental feature
+
+Validations:
+✅ No CloudFormation drift detected
+❌ Change set analysis failed
+  │ Resources: MyTableReplicauswest285A33668 (Custom::DynamoDBReplica)
+  └   • RemovalPolicy: 'Delete' (expected: 'Retain')
+
+Error: Construct upgrade validation failed
+```
+
+### Custom Validations (Optional Future Enhancement)
+
+While the currently supported constructs don't require custom validations, future upgrades
+(e.g., for AWS::EKS resources) may benefit from this feature. Additionally, this functionality would allow
+customers to incorporate enterprise-specific requirements into the validation process.
+
+For instance, users upgrading to TableV2 might want to ensure their existing table construct has
+`DeletionProtection` enabled. To accommodate such scenarios, the `cdk construct-upgrade` CLI command would
+support custom analysis files as input.  For more details about the setup and file format for custom
+validations, refer to the [Custom Validations](#custom-validations-optional-future-enhancement) section.
+
+Example usage:
+
+```sh
+cdk construct-upgrade --unstable=construct-upgrade --target aws-cdk-lib.aws-dynamodb.TableV2 \ 
+  --custom-analysis table-custom-analysis.ts
+```
+
+If the custom validation detects that `DeletionProtection` is not enabled on existing table construct,
+the CLI would display:
+
+```s
+Performing construct upgrade validation...
+
+⚠️  Note: This is an experimental feature
+
+Validations:
+✅ No CloudFormation drift detected
+✅ Change set analysis complete
+❌ Custom validation failed
+  │ Resource: MyTable794EDED1 (AWS::DynamoDB::Table)
+  └   • DeletionProtection: 'false' (expected: 'true')
+
+Error: Construct upgrade validation failed
+```
+
+This feature would enhance the flexibility of the construct upgrade process, allowing teams to enforce
+specific policies or best practices during migrations.
+
+> If you're interested in seeing a full example of VPC V2 example, please checkout
+> [Appendix B: VPC V2 Upgrade Example](#appendix-b-vpc-related-migration-example).
+
+### User Experience: Visual Workflow
```

Comment:
If you are going to show a visual with named blocks, I think you need to explain what each block is before or provide a key with links to where the explanation is. Ensure explanation covers both why it is needed and how it works.

Reply:
Thanks Simon, I'll move it to the later section after each individual validations are explained.